### ISSUE TITLE
TypeError: path.charCodeAt is not a function in getEncodedRootLength

### ISSUE THREAD

**Issue by xBoyMinemc** [microsoft/TypeScript#51518]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.73.0 %8fa188b2b301d36553cbc9ce1b0a146ccb93351f %x64  

- OS Version:  21H2 for home %19044.1889  

Steps to Reproduce:  

1. open my work,some ts file
2. Press ctrl and click on some variable, or just mouse over, to see information about the variable (Indefinite time, sometimes a few minutes, sometimes longer)  
3. ![image](https://user-images.githubusercontent.com/86919167/200857654-275ad924-5dbb-4c14-9228-e9be24482f25.png)
4. Prompt to crash five times, no more restarts

normal ： ![image](https://user-images.githubusercontent.com/86919167/200858799-b5765e15-46bd-4580-9ba7-e3fc4f95f801.png)  

crash ： 
![image](https://user-images.githubusercontent.com/86919167/200859400-539dded2-d032-4227-aeff-e2c8a648d6eb.png)  
![image](https://user-images.githubusercontent.com/86919167/200859941-7d2bf88b-c793-4b2f-93a7-10aec2fec14d.png)

**Comment by xBoyMinemc** [comment]
[log.zip](https://github.com/microsoft/TypeScript/files/10009810/log.zip)

Personal computers, theoretically do not contain sensitive information

**Comment by xBoyMinemc** [comment]
and i think,,,,,
![image](https://user-images.githubusercontent.com/86919167/201840485-cd0b2e76-f4ad-4c96-81dc-2285631a43d5.png)

CPU usage is high because log files are created continuously and some programs try to restart

**Comment by DanielRosenwasser** [comment]
This is because [your package has a `tsconfig.json` with unterminated `baseUrl` and `typeRoots` properties](https://unpkg.com/browse/@xboyminemc/server-plus@0.0.1/tsconfig.json):

```jsonc
{
    "compilerOptions": {
        "module": "commonjs",
        "lib": ["es6"],
        "target": "es6",
        "forceConsistentCasingInFileNames": true,
        "noEmit": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strictFunctionTypes": true,
        "strictNullChecks": true,
        "baseUrl": "..,
        "typeRoots": ["..],
        "types": []
    },
    "files": [
        "index.d.ts"
    ]
  }
```

On our side the fix is easy - we need to validate that when we check `option.isFilePath` that the current value is actually a `string`.

```diff
-    if (option.isFilePath) {
+    if (option.isFilePath && typeof value === "string") {
```

**Comment by DanielRosenwasser** [comment]
This is still a bug in our compiler - please leave this open until it's resolved.

**Comment by iisaduan** [comment]
The crash was fixed with #53331

**Comment by sunnywilson93** [comment]
Upgrade `typescript` to `v5+`.
 Apart from this, if you're using any bundlers (e.g., Rollup, Webpack), don't forget to upgrade their TypeScript-related packages as well. 
I had to upgrade `rollup-plugin-ts`.
